build/org.eclipse.cdt.managedbuilder.gnu.ui/src/org/eclipse/cdt/managedbuilder/gnu/ui/GnuAsmFlagsCommandGenerator.java

```diff
@@ -0,0 +1,97 @@
+/*******************************************************************************
+ * Copyright (c) 2024 John Dallaway and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    John Dallaway - initial implementation (#666)
+ *******************************************************************************/
+package org.eclipse.cdt.managedbuilder.gnu.ui;
+
+import java.util.regex.Pattern;
+
+import org.eclipse.cdt.core.cdtvariables.CdtVariableException;
+import org.eclipse.cdt.managedbuilder.core.BuildException;
+import org.eclipse.cdt.managedbuilder.core.IConfiguration;
+import org.eclipse.cdt.managedbuilder.core.IOption;
+import org.eclipse.cdt.managedbuilder.core.IOptionCommandGenerator;
+import org.eclipse.cdt.managedbuilder.core.ITool;
+import org.eclipse.cdt.managedbuilder.internal.buildmodel.FileMacroExplicitSubstitutor;
+import org.eclipse.cdt.managedbuilder.internal.macros.BuildfileMacroSubstitutor;
+import org.eclipse.cdt.utils.cdtvariables.CdtVariableResolver;
+import org.eclipse.cdt.utils.cdtvariables.IVariableSubstitutor;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.Status;
+
+/**
+ * Assembler flags command generator.
+ * This command generator supports managed build projects that were
+ * created using older versions of the GNU toolchain build description
+ * where assembly files were built by invoking the GNU "as" tool directly.
+ * @noextend This class is not intended to be subclassed by clients.
+ * @noinstantiate This class is not intended to be instantiated by clients.
+ * @since 8.7
+ */
+public class GnuAsmFlagsCommandGenerator implements IOptionCommandGenerator {
+
+	private static final String DO_NOT_LINK_FLAG = "-c"; //$NON-NLS-1$
+	private static final Pattern DO_NOT_LINK_PATTERN = Pattern.compile("(^|\\s)-c($|\\s)"); //$NON-NLS-1$
+	private static final Pattern ASM_FLAG_PATTERN = Pattern.compile("-[aDKLR]\\S*"); //$NON-NLS-1$
+
+	@Override
+	public String generateCommand(IOption option, IVariableSubstitutor macroSubstitutor) {
+		String toolCommand = getToolCommand(option, macroSubstitutor);
+		try {
+			if (null != toolCommand && IOption.STRING == option.getValueType() && option.getCommand().isEmpty()) {
+				String optionValue = option.getStringValue();
+				if (toolCommand.equals("gcc")) { //$NON-NLS-1$
```

Comment:
Sorry to be late the party here, but should this line not match `arm-non-eabi-gcc` for example, or `gcc.exe`? Or is it enough to have a strict check for `gcc`?

Reply:
We are checking only whether the tool command has been overridden. We don't care if it has been overridden by the user or by an extender. If the tool command has been overridden, then our new default value (`gcc`) will have no effect on the build and we do not need to modify the assembler flags.